src/common/harvest/harvest.js

```diff
@@ -51,87 +39,88 @@ export class Harvest extends SharedContext {
   /**
    * Initiate a harvest from multiple sources. An event that corresponds to the endpoint
    * name is emitted, which gives any listeners the opportunity to provide payload data.
+   * Note: Used by page_action
    * @param {NetworkSendSpec} spec Specification for sending data
    */
-  sendX (spec) {
-    const { endpoint, opts } = spec
-    var submitMethod = getSubmitMethod(endpoint, opts)
-    if (!submitMethod) return false
-    var options = {
-      retry: submitMethod.method === submitData.xhr
+  sendX (spec = {}) {
+    const submitMethod = submitData.getSubmitMethod({ isFinalHarvest: spec.opts?.unload })
```

Comment:
Will this always be valid and defined now? (old behavior had a case where submitMethod was being checked for existence)

Reply:
Yea, `getSubmitMethod` will now always return a submit method.